sorachung/StoriesByUs
# Src browse

#### Changes Made

1. Added file `Bookmark.cs` to `Models/` directory.
2. Added file `Chapter.cs` to `Models/` directory.
3. Added file `Genre.cs` to `Models/` directory.
4. Added file `Rating.cs` to `Models/` directory.
5. Added file `Story.cs` to `Models/` directory.
6. Added file `Tag.cs` to `Models/` directory. Models added for story card displaying purposes. 
7. Added file `IStoryRepository.cs and StoryRepository.cs` to `Repositories/` directory. StoryRepository created with GetAll method including data for all connected models.
8. Added file `StoryController.cs` to `Controllers/` directory. Has Get action/method
9. Added file `Browse.js` to `client/src/components/browse/` directory. List of links to different browsing options. Only has browse all for now
10. Added file `BrowseAll.js` to `client/src/components/browse/` directory. This component calls StoryCard to list all stories.
11. Added file `StoryCard.js` to `client/src/components/stories/` directory. Represents each story in a card layout. Hopefully reuseable.
12. Added file `storyManager.js` to `client/src/modules/` directory. Used for fetch calls related to stories.

1. Modified file `SQL/02_StoriesByUs_SeedData.sql` to include `more data to seed and to fix dates`.
2. Modified file `UserController.cs` to include `[Authorize]`.
3. Modified file `ApplicationViews.js` to include `browse routes`.
4. Modified file `Navbar.js` to include `Browse menu option`.
​

#### Steps to Review

1. Checkout this branch locally.
   ```
   git fetch --all
   git checkout src-browse
   ```
13. Open a new Terminal tab (⌘T) and navigate to the server directory.
14. Test app functionality.
   > log in
   > click Browse in menu navbar
   > click browse all
   > view the page
15. View code file.
   > Confirm file modifications are present as indicated above.
   > Confirm no unused code or extraneous comments exist.
